### PR TITLE
[CELEBORN-1196] Slots allocator will increment disk index repeatedly

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### What changes were proposed in this pull request?

Fix a bug about incorrect diskIndex calculation

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

PASS GA

### CODE REVIEW

File: master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java [hunk 1]
```diff
@@ -218,7 +218,6 @@ private static StorageInfo getStorageInfo(
                 diskInfos[diskIndex].mountPoint(),
                 diskInfos[diskIndex].storageType(),
                 availableStorageTypes);
-        diskIndex = (diskIndex + 1) % diskInfos.length;
         workerDiskIndex.put(selectedWorker, (diskIndex + 1) % diskInfos.length);
```
Comment (AngersZhuuuu) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java:221]:
```
 diskIndex = (diskIndex + 1) % diskInfos.length;
workerDiskIndex.put(selectedWorker, diskIndex);
```
?

File: master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java [hunk 2]
```diff
@@ -219,7 +219,7 @@ private static StorageInfo getStorageInfo(
                 diskInfos[diskIndex].storageType(),
                 availableStorageTypes);
         diskIndex = (diskIndex + 1) % diskInfos.length;
-        workerDiskIndex.put(selectedWorker, (diskIndex + 1) % diskInfos.length);
+        workerDiskIndex.put(selectedWorker, diskIndex);
```
Comment (pan3793) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java:222]: is the change correct? `diskIndex` was updated before the second time calculation

Comment (jiaoqingbo) [reply master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java:222]: There should be no problem. StorageInfo has been constructed successfully. workerDiskIndex uses diskIndex to record the location of the next Disk.

File: master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java [hunk 3]
```diff
@@ -219,7 +219,7 @@ private static StorageInfo getStorageInfo(
                 diskInfos[diskIndex].storageType(),
                 availableStorageTypes);
         diskIndex = (diskIndex + 1) % diskInfos.length;
```
Comment (FMX) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java:221]: Just deleting line 221 will be better and the code style will be consistent with the rest.

**Comment by waitinfuture** [comment]
Seems this is a bug introduced by https://github.com/apache/incubator-celeborn/pull/2098 ? @FMX

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.5.0) and branch 0.4(v0.4.0).

**Pull request closed without merging**